lua/nvim-web-devicons.lua

```diff
@@ -833,17 +838,25 @@ return {
       return by_name, get_highlight_name(icon_data)
     else
       icon_data = icons[ext]
-      if opts and opts.default and not icon_data then
+
+      if (global_opts.default or (opts and opts.default)) and not icon_data then
         icon_data = default_icon
       end
+
       if icon_data then
         local by_ext = icon_data.icon
         return by_ext, get_highlight_name(icon_data)
       end
     end
   end,
-  setup = function(user_icons)
-    icons = vim.tbl_extend("force", icons, user_icons or {});
+  setup = function(opts)
+    local user_icons = opts or {}
+
+    if global_opts.default then
```

Comment:
i believe this should be user_icons.default to override global_opts right ?

Reply:
oh yeah, my bad. Will fix it in a sec

Reply:
ok done.